### DISCUSSION TITLE
Two Attract Mode Feature Ideas

### DISCUSSION THREAD

**Discussion by EVILEDACCOUNT** [ikemen-engine/Ikemen-GO#873] [Feature Request]
Hello. I have two questions about Attract Mode. These can also be feature suggestions if there is no way to do them currently.

I am thinking about actually building an Arcade Machine and I would like a working coin door and the whole Attract Mode idea to work with it.

1: Is there a way to assign joystick input to the coin insertion command?

I ask this because currently coin insertion seems to be handled as a key press. However, my encoder boards handle input as an x-input joystick. If joystick input cannot currently be used, then there is no real way to wire the coin door to my encoder; because my encoder uses x-input instead of keyboard input. For an engine that works with both keyboard input as well as joystick input; it makes sense that the coin insertion command should also be able to work with joystick input.

Is this something I can do now? Maybe by replacing the key command of ‘5’ with something like ‘p1.joystick.10’ or something like that?

2: There should be a way to set a Credit Requirement.

Typically, arcades allow for the setting and the text display of how many credits are required to play. For instance, 1 Credit, 2 Credits, 3 Credits, 4 Credits, etc. Whatever number the arcade operator sees fit.

If there isn’t already, there really should be a ‘Credit Requirement’ field in system.def that allows me to put the number that I want to put in for this setting. Then, display text should show the player how many credits are needed for play out of total credits inserted.

So, something like. ‘Credits 1, Credits needed per play is 2.’

I could see this being something useful because many arcade machines accept a credit as $0.25 and now days most people want games to cost somewhere between $0.50 and $2.00 a play. Considering inflation has made $0.25 not a lot of money and that amount would probably not even pay for a public facing Cabinet to stay in good repair and operational, let alone leave any room for a real profit margin.

I love that Attract Mode exists. It has long been a highly requested feature. But, these are the two things that I think need added to it (if they’re not already there) if it’s going to be of full use to myself and others.

**Comment by Windblade-GR01** [comment]
Currently there is not a way to assign the "coin" key to a joystick.

These enhancements sounds good.
I'll work on implementing these.

**Comment by EVILEDACCOUNT** [comment]
Thank you.
